src/io/procops.c

```diff
@@ -988,17 +988,20 @@ MVMObject * MVM_proc_spawn_async(MVMThreadContext *tc, MVMObject *queue, MVMObje
         MVM_exception_throw_adhoc(tc,
             "spawnprocasync target queue must have ConcBlockingQueue REPR");
 
-    /* Encode arguments, taking first as program name. */
+    /* Encode program name and arguments. */
     arg_size = MVM_repr_elems(tc, argv);
     if (arg_size < 1)
         MVM_exception_throw_adhoc(tc, "spawnprocasync must have first arg for program");
-    args = MVM_malloc((arg_size + 1) * sizeof(char *));
-    for (i = 0; i < arg_size; i++) {
+
+    REPR(argv)->pos_funcs.at_pos(tc, STABLE(argv), argv, OBJECT_BODY(argv), 0, &reg, MVM_reg_obj);
+    prog = MVM_string_utf8_c8_encode_C_string(tc, MVM_repr_get_str(tc, reg.o));
+
+    args = MVM_malloc((arg_size - 1 + 1) * sizeof(char *));
```

Comment:
Why `arg_size - 1 + 1`? That will end up as `arg_size` anyway.

Reply:
I tried to be as explicit as possible: arg_size - program name + null marker. The compiler will most certainly optimize it away. And it already managed to trip you up and make you aware that something out of the ordinary is going on.
I don't have strong emotions though. So should you object, I can change that.